src/flb_oauth2.c

```diff
@@ -392,6 +392,8 @@ char *flb_oauth2_token_get(struct flb_oauth2 *ctx)
             flb_http_client_destroy(c);
             return ctx->access_token;
         }
+        ctx->issued = time(NULL);
+        ctx->expires = ctx->issued + ctx->expires_in;
```

Comment:
I think you want to do this in the success case, right? Just before the `return` in line 393?

Reply:
Updated